pkg/transport/generic_transport_controller.go

```diff
@@ -438,6 +435,31 @@ func (c *genericTransportController) initializeWrappedObject(ctx context.Context
 	return wrappedObject, nil
 }
 
+func (c *genericTransportController) propagateWrappedObjectToClusters(ctx context.Context, desiredWrappedObject *unstructured.Unstructured,
+	currentWrappedObjectList *unstructured.UnstructuredList, destinations []v1alpha1.Destination) error {
+	// if the desired wrapped object is nil, that means we should not propagate this object.
+	// this may happen when the workload section is empty.
+	// this is not an error state but a valid scenario.
+	// return without propagating, the delete section will remove existing instances of the wrapped object from all current destinations.
+	if desiredWrappedObject == nil {
+		return nil // this is not considered an error.
```

Comment:
What about deleting the existing wrapped objects, if any, for this BindingPolicy?

Reply:
Oh, I see, the trimming of `currentWrappedObjectList` is suppressed in this case.